datajoint/relational_operand.py

```diff
@@ -569,7 +575,7 @@ def make_argument_subquery(arg):
         """
         Decide when a Join argument needs to be wrapped in a subquery
         """
-        return Subquery.create(arg) if isinstance(arg, (GroupBy, Projection)) else arg
+        return Subquery.create(arg) if isinstance(arg, (GroupBy, Projection)) or arg.restriction else arg
```

Comment:
This will work but with a potential slow downs as there could be cases where restriction can be passed through without a need for a subquery, and thus this can be a slow down.

Reply:
true.  We value correctness over efficiency.  however, we will soon release a new optimization redesign that's based on slightly different overall approach.

Reply:
Between correctness and efficiency, there is no question asked - we always strive for correctness, and there should be no exception to this. This will be a notable optimization point as subquery is known to dramatically decrease query speed. I don't know what different approach you are thinking of, but I agree that this is not the only place that needs attention on query optimization, and might be better dealt together. I'll likely create an issue on this, so we can track improvements in the future.